internal/controller/apiextensions/composite/composition_functions.go

```diff
@@ -66,6 +71,7 @@ const (
 	errFmtUnmarshalDesiredCD         = "cannot unmarshal desired composed resource %q from RunFunctionResponse"
 	errFmtCDAsStruct                 = "cannot encode composed resource %q to protocol buffer Struct well-known type"
 	errFmtFatalResult                = "pipeline step %q returned a fatal result: %s"
+	errFmtFunctionMaxIterations      = "pipeline step %q reached the maximum number of iterations (%d)"
```

Comment:
Similar to render, I think it would be useful to explain that we hit max iterations because the function's requirements didn't stabilize.